test/bats/provider.bats

```diff
@@ -0,0 +1,164 @@
+    fi
+
+    { # Braces used to redirect all teardown logs.
+    # Teardown Vault configuration.
+    kubectl --namespace=csi exec vault-0 -- vault auth disable kubernetes
+    kubectl --namespace=csi exec vault-0 -- vault policy delete example-readonly
+    kubectl --namespace=csi exec vault-0 -- vault kv delete secret/foo1
+    kubectl --namespace=csi exec vault-0 -- vault kv delete secret/foo2
+    kubectl --namespace=csi exec vault-0 -- vault kv delete secret/foo-sync1
+    kubectl --namespace=csi exec vault-0 -- vault kv delete secret/foo-sync2
+
+    # Teardown shared k8s resources.
+    kubectl delete --ignore-not-found namespace test
+    kubectl delete --ignore-not-found namespace negative-test-ns
+    } > $SETUP_TEARDOWN_OUTFILE
+}
+
+@test "1 Inline secrets-store-csi volume" {
+    kubectl --namespace=test apply -f $CONFIGS/nginx-inline-volume.yaml
+    kubectl --namespace=test wait --for=condition=Ready --timeout=60s pod/nginx-secrets-store-inline
+
+    result=$(kubectl --namespace=test exec nginx-secrets-store-inline -- cat /mnt/secrets-store/bar1)
+    [[ "$result" == "hello1" ]]
+
+    result=$(kubectl --namespace=test exec nginx-secrets-store-inline -- cat /mnt/secrets-store/bar2)
+    [[ "$result" == "hello2" ]]
+}
+
+@test "2 Sync with kubernetes secrets" {
+  # Deploy some pods that should cause k8s secrets to be created.
+  kubectl --namespace=test apply -f $CONFIGS/nginx-env-var.yaml
+  kubectl --namespace=test wait --for=condition=Ready --timeout=60s pod -l app=nginx
+
+  POD=$(kubectl --namespace=test get pod -l app=nginx -o jsonpath="{.items[0].metadata.name}")
+  result=$(kubectl --namespace=test exec $POD -- cat /mnt/secrets-store/bar1)
+  [[ "$result" == "hello-sync1" ]]
+
+  result=$(kubectl --namespace=test exec $POD -- cat /mnt/secrets-store/bar2)
+  [[ "$result" == "hello-sync2" ]]
+
+  result=$(kubectl --namespace=test get secret foosecret -o jsonpath="{.data.pwd}" | base64 -d)
+  [[ "$result" == "hello-sync1" ]]
+
+  result=$(kubectl --namespace=test exec $POD -- printenv | grep SECRET_USERNAME | awk -F"=" '{ print $2 }' | tr -d '\r\n')
+  [[ "$result" == "hello-sync2" ]]
+
+  run kubectl get secret --namespace=test foosecret
+  [ "$status" -eq 0 ]
+
+  result=$(kubectl --namespace=test get secret foosecret -o jsonpath="{.metadata.labels.environment}")
+  [[ "${result//$'\r'}" == "test" ]]
+
+  result=$(kubectl --namespace=test get secret foosecret -o jsonpath="{.metadata.labels.secrets-store\.csi\.k8s\.io/managed}")
+  [[ "${result//$'\r'}" == "true" ]]
+
+  # There isn't really an event we can wait for to ensure this has happened.
+  for i in {0..60}; do
+      result="$(kubectl --namespace=test get secret foosecret -o json | jq '.metadata.ownerReferences | length')"
+      if [[ "$result" -eq 2 ]]; then
+          break
+      fi
+      sleep 1
+  done
+  [[ "$result" -eq 2 ]]
+
+  # Wait for secret deletion in a background process.
+  kubectl --namespace=test wait --for=delete secret foosecret &
```

Comment:
Should this include a timeout arg? Unless 30s is enough (looks like that's the default).

Reply:
Good spot, I've tried to be consistently generous with timeouts to avoid flakes, so I've added 60s as with the other waits.